README.md

```diff
@@ -4,47 +4,97 @@
 uftrace
 =======
 
-The uftrace tool is to trace and analyze execution of a program
-written in C/C++.  It was heavily inspired by the ftrace framework
-of the Linux kernel (especially function graph tracer) and supports
-userspace programs.  It supports various kind of commands and filters
-to help analysis of the program execution and performance.
+uftrace is a function call graph tracer for C, C++, Rust and Python programs.
+Using dynamic tracing, it can include nested library calls and
```

Comment:
Thank at lot, and according to `doc/uftrace-live.md`, it also supports `-mfentry`, `-mnop-mcount`, and using clang, `-fxray-instrument`.

With this information, I've updated the start of the README like this now:

> It can record data from:
> - User space functions by dynamically patching functions at runtime, or with code
  compiled with with `-pg`, `-fpatchable-function-entry`, `-mfentry`, `-mnop-mcount`,
  `-finstrument-functions`, or using clang with `-fxray-instrument`.
>- Library functions (through PLT hooking)
>- Kernel functions (using the ftrace framework in Linux kernel)
>- Kernel trace events affecting the program, e.g. scheduling events (which affect the
  execution timing of the program) and records nanosecond-exact timestamps
  (using perf_event and systemtap SDT)
>
>With the recorded data, it:
>- Shows colored nested function call graphs (instead of listings)!
>- Displays arguments symbolically using libc function prototypes and DWARF
  debug information values.
>- Can apply filters like minimal function call duration
>- Can display metadata about the system on which the trace was taken
>- Can generate a symbol table of the traced program and library functions
>- When the execution of multiple programs was recorded,
  it can generate a task relationship tree (parent and childs) from the trace.

Reply:
Looks great! Thanks.